server/bleep/src/cache.rs

```diff
@@ -0,0 +1,280 @@
+use std::sync::{Arc, RwLock};
+
+use qdrant_client::{
+    prelude::QdrantClient,
+    qdrant::{
+        points_selector::PointsSelectorOneOf, with_payload_selector, with_vectors_selector, Filter,
+        PointId, PointStruct, PointsIdsList, PointsSelector, ScrollPoints, WithPayloadSelector,
+        WithVectorsSelector,
+    },
+};
+use sqlx::Sqlite;
+use uuid::Uuid;
+
+use crate::{
+    repo::RepoRef,
+    semantic::{self, make_kv_keyword_filter, Embedding, Payload},
+};
+
+use super::db::SqlDb;
+
+#[derive(serde::Serialize, serde::Deserialize, Eq)]
+pub(crate) struct FreshValue<T> {
+    // default value is `false` on deserialize
+    pub(crate) fresh: bool,
+    pub(crate) value: T,
+}
+
+impl<T> PartialEq for FreshValue<T>
+where
+    T: PartialEq,
+{
+    fn eq(&self, other: &Self) -> bool {
+        self.value.eq(&other.value)
+    }
+}
+
+impl<T> FreshValue<T> {
+    fn stale(value: T) -> Self {
+        Self {
+            fresh: false,
+            value,
+        }
+    }
+}
+
+impl<T> From<T> for FreshValue<T> {
+    fn from(value: T) -> Self {
+        Self { fresh: true, value }
+    }
+}
+
+pub(crate) type RepoCacheSnapshot = Arc<scc::HashMap<String, FreshValue<()>>>;
```

Comment:
Can you add comments explaining what these different levels of caching do?
- `RepoCacheSnapshot`
- `FileCache`
- `ChunkCache`